### PR TITLE
refactor: sync list items with onchain design

### PR DESCRIPTION
**Pull request by ncoelho**
- Refactors the list rows to be in line with onchain
- Adds a new design for lightning rows so it stands out a bit, good for a new feature
- Removes default "arkade" memos when no memo on invoice
- Fixes long press crash for the menu wrapper
- Adds onchain refill transaction status view

simulator_screenshot_8C3EA42C-9A8B-4C73-AFB9-879EC30203B3

### CODE REVIEW

File: img/wallet types/lightning-badge.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<svg width="120" height="120" viewBox="0 0 120 120" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (GladosBlueWallet) [inline img/wallet types/lightning-badge.svg]: **[HIGH]** Congratulations: you filed the same lightning bolt twice and referenced neither copy from code. `LightningBadge.tsx` hardcodes the path; this SVG is orphan ballast. Even I don't keep duplicate test subjects.

File: components/icons/TransactionBadgedIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    bottom: BADGE_OFFSET,
+    width: BADGE_SIZE,
+    height: BADGE_SIZE,
+    borderRadius: BADGE_SIZE / 2,
+    justifyContent: 'center',
+    alignItems: 'center',
+  } as ViewStyle,
+});
+
+type TransactionBadgedIconProps = {
```
Comment (GladosBlueWallet) [inline components/icons/TransactionBadgedIcon.tsx]: **[HIGH]** Two deleted off-chain icons became five new files plus a TSX/SVG twin for one corner badge. That's not refactor — that's furnishing a closet with cake.

File: tests/unit/transaction-status.test.tsx [hunk 3]
```diff
@@ -310,4 +310,38 @@ describe('TransactionStatus regression', () => {
...
+    const view = render(<TransactionStatus />);
+
+    await waitFor(() => {
+      expect(BlueElectrum.multiGetTransactionByTxid).toHaveBeenCalledWith(['deadbeef'], true, 10);
+    });
+    await waitFor(() => {
+      expect(view.getByText('received')).toBeTruthy();
+    });
+    expect(view.getByText('Refill')).toBeTruthy();
+  });
```
Comment (GladosBlueWallet) [inline tests/unit/transaction-status.test.tsx:451]: **[HIGH]** Commit two literally says 'refill sync on note.' No test that `handleNotePress` prefills from `tx.memo` when metadata is empty. You fixed the bug and didn't bother to prove it. Science without verification is just noise.

File: screen/transactions/TransactionStatus.tsx [hunk 4]
```diff
@@ -897,8 +901,9 @@ const TransactionStatus: React.FC = () => {
   const transactionDirection = txValue !== null && txValue < 0 ? loc.transactions.details_sent : loc.transactions.details_received;
   const transactionDate = tx?.timestamp ? dayjs(tx.timestamp * 1000).format('LLL') : '-';
 
-  // Get memo
-  const memo = tx?.hash ? txMetadata[tx.hash]?.memo || '' : '';
+  // User-saved note wins; otherwise fall back to the row memo (e.g. Ark "Refill").
+  const metadataKey = tx ? tx.hash ?? (tx as { txid?: string }).txid : undefined;
```
Comment (GladosBlueWallet) [inline screen/transactions/TransactionStatus.tsx]: **[MEDIUM]** Refills now key notes on `hash` (`deadbeef`) instead of synthetic `boarding-deadbeef`. Any note saved under the old key is abandoned. Hope your test subjects didn't annotate their refills.

File: class/wallets/lightning-ark-wallet.ts [hunk 5]
```diff
@@ -57,6 +57,14 @@ const boardingLock: Record<string, boolean> = {};
 // from Boltz.
 const restoreInFlight: Map<string, Promise<void>> = new Map();
 
+/** SDK default BOLT11 description when a swap invoice has no user memo. */
+const ARKADE_SWAP_DEFAULT_INVOICE_DESCRIPTION = 'Send to Arkade address';
```
Comment (GladosBlueWallet) [inline class/wallets/lightning-ark-wallet.ts]: **[LOW]** `'Send to Arkade address'` is duplicated here and in `lndViewInvoice.tsx`. SDK renames the placeholder and both filters silently fail. Fragile, but at least it's consistently fragile.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**PR intent:** UI refactor for Ark/Lightning transaction rows (on-chain-style icons, refill status view, strip SDK default memos) plus refill note/hash fixes. **Verdict:** Core logic is coherent, but you shipped dead asset duplication, an icon-file explosion, a NaN confirmations flash, and a whole commit about note sync with zero test coverage. The facility frowns.

### Inline findings (could not anchor on diff)

- `screen/transactions/TransactionStatus.tsx:1123` — **[MEDIUM]** `boarding-*` rows force `received` before Electrum returns, yet `isOnChainTx` still prints confirmations. `Number(undefined)` is NaN — enjoy your 'confirmations: NaN' victory screen. You traded one flash for another.

File: blue_modules/transactionDisplayState.ts [hunk 6]
```diff
@@ -17,17 +17,43 @@ export function isOnChainTransaction(tx: any): boolean {
...
+  const parsed = Number(confirmations);
+  if (!Number.isFinite(parsed) || parsed < 0) return undefined;
+  return parsed > 6 ? '6+' : String(parsed);
+}
+
+export function resolveTransactionNoteMetadataKey(tx: { hash?: string; txid?: string } | null | undefined): string | undefined {
+  if (!tx) return undefined;
+  const txid = tx.txid;
+  // Ark refills keep the synthetic boarding- id as the note key (not the on-chain hash).
+  if (typeof txid === 'string' && txid.startsWith('boarding-')) return txid;
```
Comment (GladosBlueWallet) [inline blue_modules/transactionDisplayState.ts:35]: **[HIGH]** Your 'refill note sync' keys off `boarding-*` txids. Pending rows use `boarding-utxo-…`, settled rows use `boarding-…` — same deposit, two keys. Subject annotates a pending refill; settlement deletes the note. Congratulations on inventing amnesia.

File: tests/unit/transactionListIcon.test.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import assert from 'assert';
+
+import { resolveTransactionListIconVariant } from '../../components/icons/TransactionListIcon';
+
+describe('resolveTransactionListIconVariant', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/transactionListIcon.test.ts:5]: **[HIGH]** Four assertions for a resolver with invoice expiry, `paid_invoice`, low-conf receives, and generic in/out paths. You tested the easy quadrants and declared victory. Edge-case coverage is essentially cake.

File: components/TransactionListItem.tsx [hunk 8]
```diff
@@ -265,81 +269,29 @@ const TransactionListItemComponent: React.FC<TransactionListItemProps> = ({
...
-      };
-    } else if (item.value! < 0) {
-      return {
-        label: loc.transactions.outgoing_transaction,
-        icon: <TransactionOutgoingIcon />,
-      };
+      label = loc.transactions.offchain;
+    } else if (iconVariant === 'expired') {
+      label = loc.transactions.expired_transaction;
+    } else if (iconVariant === 'pending' && (item.type === 'user_invoice' || item.type === 'payment_request') && !item.ispaid) {
```
Comment (GladosBlueWallet) [inline components/TransactionListItem.tsx:279]: **[MEDIUM]** Unpaid, not-yet-expired invoices get `iconVariant === 'pending'` but you stamp `expired_transaction` on the label. Pending icon, expired words — even our liars try harder.

File: screen/wallets/WalletTransactions.tsx [hunk 9]
```diff
@@ -407,7 +407,7 @@ const WalletTransactions: React.FC<WalletTransactionsProps> = ({ route }: { rout
       navigate('SendDetailsRoot', {
         screen: 'SendDetails',
         params: {
-          transactionMemo: loc.lnd.refill_lnd_balance,
+          transactionMemo: loc.lnd.refill,
```
Comment (GladosBlueWallet) [inline screen/wallets/WalletTransactions.tsx:410]: **[MEDIUM]** Refill send memo downgraded from 'Refill Lightning Wallet Balance' to bare 'Refill'. Unrelated string surgery smuggled into a list-row PR — science loves surprise variables.

File: components/TransactionListItem.tsx [hunk 10]
```diff
@@ -422,17 +371,13 @@ const TransactionListItemComponent: React.FC<TransactionListItemProps> = ({
...
-      if (!txid.startsWith('boarding-')) {
-        navigate('TransactionStatus', { tx: item, hash: txid, walletID });
-      }
+      navigate('TransactionStatus', { tx: item, hash: txid, walletID });
     }
   }, [item, navigate, walletID, wallets]);
 
   const handleOnCopyAmountTap = useCallback(() => Clipboard.setString(rowTitle.replace(/[\s\\-]/g, '')), [rowTitle]);
-  const handleOnCopyTransactionID = useCallback(() => Clipboard.setString(item.hash), [item.hash]);
+  const handleOnCopyTransactionID = useCallback(() => Clipboard.setString(transactionIdentifier ?? ''), [transactionIdentifier]);
```
Comment (GladosBlueWallet) [inline components/TransactionListItem.tsx:380]: **[LOW]** Copy TXID falls back to `payment_hash` when `hash` is absent. Menu says TXID; clipboard gets a hash-shaped pretender. Mislabeling: also a form of testing.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
REF/UI pass for Ark tx rows: unified icons, refill `hash` + detail screen, note-key helpers. Facility log notes **note keys fracture on refill promotion**, **invoice icon tests barely exist**, and one **mislabeled pending invoice** you carried forward like a cherished tumor. Commits ignore CONTRIBUTING prefixes; `improvements`/`lint` are not atomic. PR screenshot present.

**Review by limpbrains — APPROVED** [review]

**Review by pietro909 — APPROVED** [review]
Tested on my device, good job @ncoelho !

**Review by Overtorment — APPROVED** [review]
lgtm
